xlators/mgmt/glusterd/src/glusterd-handler.c

```diff
@@ -164,12 +164,12 @@ glusterd_handle_friend_req(rpcsvc_request_t *req, uuid_t uuid, char *hostname,
         goto out;
     }
 
-    event->peername = gf_strdup(peerinfo->hostname);
+    gf_strncpy(event->peername, peerinfo->hostname, sizeof(event->peername));
```

Comment:
If for some reason the source hostname is larger than `GF_HOST_NAME_MAX`, shouldn't we fail with an error instead of silently truncating the hostname, which could have unwanted side effects ?